### ISSUE TITLE
A question

### ISSUE THREAD

**Issue by XahidEx** [JuzouSatoru2/JuzouSatoru2#1]
bro, were you able to deploy sanity v3? I've tried enough. still getting issues. 
`sanity.cli.ts does not contain a project identifier ("api.projectId"), which is required for the Sanity CLI to communicate with the Sanity API`
even though I've specified projectId.

**Comment by alex289** [comment]
Yes, I got the same error. Do you have a public repo I can look at?

**Comment by alex289** [comment]
I think the problem was the missing `sanity.cli.ts` and updating some deps. [Here is my project](https://github.com/alex289/Portfolio), maybe you can find what is different in your project

**Comment by XahidEx** [comment]
I did.

> Yes, I got the same error. Do you have a public repo I can look at?

No public repo. But I followed leerob mostly. And when I wanted to deploy sanity v3 studio I got this error. I also followed your repo if I missed something. But still I couldn't fix it.
sanity.cli.ts also exist. + updated packages following your repo. Still couldn't. I have a question, How do you access the studio? How do you assign your username for studio url? (Using vercel)

**Comment by alex289** [comment]
I just run `pnpm sanity deploy` and on a first run sanity asks what name you want to deploy it to I think.

There must be both the sanity.config.ts and the sanity.cli.ts file in the projects root.

Beside the plugins I just have sanity@3.0.0-dev-preview.17 installed

**Comment by alex289** [comment]
Did you get any further? Sanity v3 got an update yesterday and maybe it got fixed or your sanity.cli.ts  does not have the correct structure.

**Comment by XahidEx** [comment]
Solved 😅... But I saw on Twitter that v3 Studio for self-hosting isn't released yet. But not a problem. I got what I wanted. ❤

**Comment by XahidEx** [comment]
![Screenshot_20220925-124633_Firefox](https://user-images.githubusercontent.com/88249321/192131767-dd8ba627-fb2c-4e3b-9311-a1f1770579aa.png)
![Screenshot_20220925-124641_Firefox](https://user-images.githubusercontent.com/88249321/192131770-cec190e9-ec21-4f02-8c1e-720bd8962a2c.png)
Do you think I made any mistakes here?
Because of demand ISR not working properly 😪

**Comment by alex289** [comment]
Good news.

About the revalidation, most of your settings look similar to mine but I have a custom projection { _id } and I hope you have setup your secret both in your project and on sanity. The last thing that may be the error is your api route. Maybe look up how I designed my revalidate hook even tho I mostly followed the tutorial from sanity itself.

**Comment by XahidEx** [comment]
I followed Leerob's repo. 😅
And yes, I set secrets in both places.

**Comment by alex289** [comment]
This [pr](https://github.com/leerob/leerob.io/pull/537) in leerobs repo has the correct setup. If this doesn't help I'd try to debug and search for what specific is not working. 😅

**Comment by XahidEx** [comment]
You just tell me one thing. What should be in these according to the leerob repo? 😅
![20220925_160543](https://user-images.githubusercontent.com/88249321/192138200-f8588f79-b8af-4cde-b516-029f1c695074.png)
I think it will be the solution for me.

**Comment by alex289** [comment]
I have set filter to `_type == 'post'` and if you have the same setup as leerob you need to set the projection to `{ _id }`. On the sanity api page you can click on the three dots and show the attempt log to verify the requests.

**Comment by XahidEx** [comment]
Thanks a lot.. & also for preview api?

**Comment by alex289** [comment]
Do you mean if it revalidates with the preview mode? On preview mode the content will be fetched on call I think so I guess it should work. Hope I could help you 😀

**Comment by alex289** [comment]
Hope I could help you. If you have still issues feel free to reopen ❤️

**Comment by alex289** [comment]
I'm sorry I never used Adsense but if you have set security headers my guess is you need to allow some adsense source in a one of the headers

**Comment by alex289** [comment]
Hey, no problem. I think are looking for something like this: your class needs a size selector like "md" and then the size "text-lg" or similar. Additionally you need to set the size for other screen sizes like "text-base".

TLDR;
Your class: "text-base md:text-lg"
Does this work?

**Comment by XahidEx** [comment]
haha, thanks a lot, bro, It worked. I was trying something else for few days.

**Comment by XahidEx** [comment]
Bro, Came up with another question. You don't have Twitter, So writing here.
I was trying to fetch Instagram API . And I was successfully able to do it on Localhost. But when I deploy it to Vercel, I face issues.  It gets blocked.  But It's working smoothly on Localhost. 
And I was searching for a solution, Somehow I found something related to CORS, which I tried. But It didn't work. Do you have any idea, what can be the solution, please?

**Comment by alex289** [comment]
Hey, Im sorry. I don't know what error you have. If it's related to CORS you maybe need to set the headers on your site and if it's still not working it may be Instagrams headers blocking you. Maybe this is a resource you can use https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS/Errors
